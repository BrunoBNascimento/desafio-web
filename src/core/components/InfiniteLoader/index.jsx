import React from 'react';
import styled from 'styled-components';

const InfiniteLoaderWrapper = styled.div`
  img{
    width:32px;
    height:32px;
    margin: auto;
    display: block;
  }
`;

const InfiniteLoader = () => (
  <InfiniteLoaderWrapper>
    <img src="https://loading.io/spinners/infinity/lg.infinity-rotate-cycle-loader.gif" alt="Loading more repositories" />
  </InfiniteLoaderWrapper>
);

export default InfiniteLoader;
