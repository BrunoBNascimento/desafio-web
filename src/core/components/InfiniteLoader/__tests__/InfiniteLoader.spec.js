import React from 'react';
import enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import InfiniteLoader from '../';

// tslint:disable-next-line:no-any
enzyme.configure({ adapter: new Adapter() });

describe('InfiniteLoader', () => {
  it('should render InfiniteLoader', () => {
    const infiniteLoader = shallow(<InfiniteLoader />);
    expect(infiniteLoader).toMatchSnapshot();
  });
});
