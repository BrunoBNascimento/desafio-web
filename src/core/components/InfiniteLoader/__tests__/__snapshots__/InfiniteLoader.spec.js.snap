// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfiniteLoader should render InfiniteLoader 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InfiniteLoader />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <img
        alt="Loading more repositories"
        src="https://loading.io/spinners/infinity/lg.infinity-rotate-cycle-loader.gif"
/>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "alt": "Loading more repositories",
        "src": "https://loading.io/spinners/infinity/lg.infinity-rotate-cycle-loader.gif",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <img
          alt="Loading more repositories"
          src="https://loading.io/spinners/infinity/lg.infinity-rotate-cycle-loader.gif"
/>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Loading more repositories",
          "src": "https://loading.io/spinners/infinity/lg.infinity-rotate-cycle-loader.gif",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
